test/ozaria/site/components/interactive/PageInteractive.spec.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-env jasmine */
+import { mount } from '@vue/test-utils'
+import pageInteractive from 'ozaria/site/components/interactive/PageInteractive/index'
+import factories from 'test/app/factories'
+import interactiveApi from 'ozaria/site/api/interactive'
+import draggableOrderingComponent from 'ozaria/site/components/interactive/PageInteractive/draggableOrdering'
+import inserCodeComponent from 'ozaria/site/components/interactive/PageInteractive/insertCode'
+import draggableStatementCompletionComponent from 'ozaria/site/components/interactive/PageInteractive/draggableStatementCompletion'
+
+const createComponent = (values = {}) => {
+  return mount(pageInteractive, {
+    propsData: values
+  })
+}
+
+const interactive = {
+  interactiveType: 'draggable-ordering',
+  promptText: 'Interactive prompt text',
+  _id: 'interactiveId1'
+}
+
+const introLevel = {
+  _id: 'intro-level-id'
+} // Dummy intro level object
+
+let pageInteractiveWrapper
+
+describe('Interactive Page', () => {
+  beforeEach((done) => {
+    spyOn(interactiveApi, 'getInteractive').and.returnValue(Promise.resolve(interactive))
+    spyOn(interactiveApi, 'getSession').and.returnValue(Promise.resolve({}))
+    me.set(factories.makeUser({ permissions: ['admin'] }).attributes)
+    pageInteractiveWrapper = createComponent({ interactiveIdOrSlug: interactive._id, introLevelId: introLevel._id })
+    _.defer(done)
```

Comment:
Why is this deferred?

Reply:
This is required so that the creation of the component is completed before we move on. The `created` function in the vue component is asynchronous so we are waiting for it to complete before we check the values in the specific tests.